var/spack/repos/builtin/packages/elemental/package.py

```diff
@@ -62,15 +65,21 @@ class Elemental(CMakePackage):
     variant('build_type', default='Release',
             description='The build type to build',
             values=('Debug', 'Release'))
+    variant('blas', default='openblas', values=('openblas', 'mkl'),
+            description='Enable the use of OpenBlas/MKL')
```

Comment:
I don't think this is the right way to do things. You should add a `+blas` variant that is either True or False. Then you should use:
```python
depends_on('blas', when='+blas')
```
If someone wants to build `elemental` with OpenBLAS, they would use:
```
$ spack install elemental+blas ^openblas
```
If they want to build with MKL, they use:
```
$ spack install elemental+blas ^mkl
```
If you want to prevent people from building with any other BLAS provider, you would add:
```python
# Only builds with OpenBLAS and MKL
conflicts('^atlas')
conflicts('^netlib-lapack')
conflicts('^veclibfort')
```
This is how every other Spack package handles virtual dependencies.

Reply:
for the record, here's the link to the issue https://github.com/LLNL/spack/issues/1712